File: src/main/java/org/gangel/graphomance/arangodb/ArangoConnectionProducer.java
package org.gangel.graphomance.arangodb;

import com.arangodb.ArangoDB;
import org.gangel.graphomance.Connection;
import org.gangel.graphomance.ConnectionProducer;
import org.gangel.graphomance.ConnectionSettings;

import java.util.Objects;

public class ArangoConnectionProducer implements ConnectionProducer {

    @Override
    public Connection connect(ConnectionSettings settings) {
        ArangoConnectionSettings connSettings = Objects.requireNonNull((ArangoConnectionSettings) settings);
        ArangoDB arangoDB = new ArangoDB.Builder()
                .user(connSettings.getUser())
                .password(connSettings.getPassword())
                .build();
        return ArangoConnection.builder()
                .db(arangoDB)
                .connectionSettings(connSettings)
                .build();
    }

}

File: src/main/kotlin/org/gangel/graphomance/arangodb/ArangoConnectionProducer.kt
package org.gangel.graphomance.arangodb

import com.arangodb.ArangoDB
import org.gangel.graphomance.Connection
import org.gangel.graphomance.ConnectionProducer
import org.gangel.graphomance.ConnectionSettings

class ArangoConnectionProducer : ConnectionProducer {
	override fun connect(settings: ConnectionSettings): Connection {
		val connSettings = settings as ArangoConnectionSettings ?: throw RuntimeException("Required ArangoDB")
		val arangoDB = ArangoDB.Builder()
			.user(connSettings.user)
			.password(connSettings.password)
			.build()
		return ArangoConnection(db = arangoDB, connectionSettings = connSettings)
	}
}